src/modules/core/filter_box_blur.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * filter_box_blur.c
+ * Copyright (C) 2011-2021 Meltytech, LLC
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include "image_proc.h"
+
+#include <framework/mlt_filter.h>
+#include <framework/mlt_frame.h>
+#include <framework/mlt_image.h>
+#include <framework/mlt_profile.h>
+
+#include <math.h>
+
+static int filter_get_image( mlt_frame frame, uint8_t **image, mlt_image_format *format, int *width, int *height, int writable )
+{
+	int error = 0;
+	mlt_filter filter =  (mlt_filter) mlt_frame_pop_service( frame );
+	mlt_properties properties = MLT_FILTER_PROPERTIES( filter );
+	mlt_position position = mlt_filter_get_position( filter, frame );
+	mlt_position length = mlt_filter_get_length2( filter, frame );
+	mlt_profile profile = mlt_service_profile( MLT_FILTER_SERVICE( filter ) );
+	double hradius = mlt_properties_anim_get_double( properties, "hradius", position, length );
+	double vradius = mlt_properties_anim_get_double( properties, "vradius", position, length );
+	// Convert from percent to pixels
+	hradius = hradius * (double)profile->width * mlt_profile_scale_width( profile, *width ) / 1000.0;
+	hradius = MAX(lrint(hradius), 0);
+	vradius = vradius * (double)profile->width * mlt_profile_scale_width( profile, *width ) / 1000.0;
```

Comment:
Use `mlt_profile_scale_height()` here

Reply:
I want the scale factor to be the same for H and V so the user does not have to guess-and-check the values if they want even h/v blur size. I refactored this code to make it more obvious.